magnetx-app-inspection/src/main/java/magnetx/InspectionDumpPlugin.kt

```diff
@@ -0,0 +1,46 @@
+package magnetx
+
+import com.facebook.stetho.dumpapp.ArgsHelper
+import com.facebook.stetho.dumpapp.DumpUsageException
+import com.facebook.stetho.dumpapp.DumperContext
+import com.facebook.stetho.dumpapp.DumperPlugin
+import magnet.Scope
+import java.io.PrintStream
+
+internal class InspectionDumpPlugin(
+    private val scope: Scope
+) : DumperPlugin {
+
+    override fun getName(): String = NAME_MAGNET
+
+    override fun dump(dumpContext: DumperContext) {
+        val writer: PrintStream = dumpContext.stdout
+        val args: Iterator<String> = dumpContext.argsAsList.iterator()
+
+        val command: String? = ArgsHelper.nextOptionalArg(args, null)
+        when (command) {
+            CMD_SCOPE -> doDumpScopes(args, writer)
```

Comment:
This is the actual meat - why have a Stetho dependency here at all if you could e.g. route the same output through a logger? Or a simple file writer? Or anything else more sophisticated? Wouldn't it be better to have this separated?

Reply:
It's not finished yet. There will be two extensions: generic stetho extension `magnetx-app-stetho` and stetho-scope dumper `magnetx-app-stetho-scope`.